### DISCUSSION TITLE
is it possible to compile libmdbx for visual studio 2008

### DISCUSSION THREAD

**Discussion by gfanini** [erthink/libmdbx#329] [Q&A] [answered]
is it possible to compile libmdbx for visual studio 2008, 32 bit, although the target vs2008 is available from cmake it won't compile

**Comment by erthink** [comment, marked as answer]
Hypothetically it is possible, but you have to solve all the problems yourself and provide testing.

I have no plans to support older versions of MSC compilers, even a little older than ones are currently stated.

**Comment by gfanini** [comment]
has anyone tried say to compile lmdbx for embedded systems say not visual studio windows or linux, I was thinking perhaps to dump the preprocessor defines prior to compilation ?
I managed to somehow compile it with visual studio 2008 which is C89 and requires declaring variables at start of scopes {}, I enclose it in case anyone say needs to use a C89 compiler say some really old old system like the one I work on ?

[liblmdbx_vs2008.zip](https://github.com/erthink/libmdbx/files/8165989/liblmdbx_vs2008.zip)

**Comment by erthink** [comment]
> has anyone tried say to compile lmdbx for embedded systems say not visual studio windows or linux, I was thinking perhaps to dump the preprocessor defines prior to compilation ?

For now libmdbx support a set of OS: Linux (including Android), *BSD (FreeBSD, OpenBSD, DragonFly, NetBSD), OSX/iOS, Solaris, Windows. To support any other platform not full compatible a one of supported a set of changes should be made.

If the new target OS is highly compatible with one of supported then a small set of trivial changes is required, for instance modify some ifdefs, etc.

On the other hand, if the new target OS is unique (i.e. incompatible with any of supported), then this can be a difficult challenge requiring knowledge both libmdbx internals and the target OS properties/API.

**Comment by gfanini** [comment]
another question : is it possible to open a write transaction from thread 1, while another thread 2 is holding a write transaction opened for multiple mdbx_put (from thread 2), they seem to deadlock on me_windowsbug_lock critical section ?
could it be possible to open multiple write transactions ?
I have to write from only a single thread single writer thread though ?

**Comment by erthink** [comment]
Definitely no, please read the readme and documentation.
However, there's no deadlock but transaction serialization.
